source/tests/SGM/SGMEnergyTrajectoryTestView.cpp

```diff
@@ -242,34 +263,20 @@ void SGMEnergyTrajectoryTestView::setupUi()
 	energyVsTimeSeries->setDescription("Energy produced by constant velocity grating angle");
 }
 
-void SGMEnergyTrajectoryTestView::setTheoreticalPlotData(SGMGratingSupport::GratingTranslation gratingTranslation)
+void SGMEnergyTrajectoryTestView::setTheoreticalPlotData(SGMGratingSupport::GratingTranslation gratingTranslation,
+                                                         SGMUndulatorSupport::UndulatorHarmonic undulatorHarmonic)
 {
 	if(gratingTranslation != SGMGratingSupport::UnknownGrating) {
-		double startValue;
-		double endValue;
-
-		switch(gratingTranslation) {
-		case SGMGratingSupport::LowGrating:
-			startValue = 200;
-			endValue = 800;
-			break;
-		case SGMGratingSupport::MediumGrating:
-			startValue = 400;
-			endValue = 1400;
-			break;
-		case SGMGratingSupport::HighGrating:
-			startValue = 600;
-			endValue = 2500;
-			break;
-		default:
-			startValue = 0;
-			endValue = 0;
-		}
+		double startValue = 200;
+		double endValue = 3000;
 
 		double increment = 1;
 
 		int numberOfItems = int((endValue - startValue) / increment);
```

Comment:
increment is the value to increment the energy by for each X point. This way if I need to test something with greater fidelity than 1eV I can alter this one number (which I did during testing of this issue to ensure the I had the value which caused the asymptote down to values of 0.001eV). A better name for this would probably be deltaEnergy.

I've cleaned up this area of code a little bit and will push up shortly.